src/pip/_internal/wheel.py

```diff
@@ -693,10 +705,33 @@ def _base_setup_args(self, req):
             SETUPTOOLS_SHIM % req.setup_py
         ] + list(self.global_options)
 
-    def __build_one(self, req, tempd, python_tag=None):
+    def _build_one_pep517(self, req, tempd, python_tag=None):
+        assert req.metadata_directory is not None
+        try:
+            req.spin_message = 'Building wheel for %s (PEP 517)' % (req.name,)
+            logger.debug('Destination directory: %s', tempd)
+            wheelname = req.pep517_backend.build_wheel(
+                tempd,
+                metadata_directory=req.metadata_directory
+            )
+            if python_tag:
+                # General PEP 517 backends don't necessarily support
+                # a "--python-tag" option, so we rename the wheel
```

Comment:
I don't like how this looks but I don't really have a better idea. :)

Reply:
It's ugly as hell, but short of a PEP update defining a standard way for frontends to ask backends to allow a particular Python tag, there's no better solution.

Whether the use cases for `--python-tag` are strong enough to warrant such a standard (or whether it could be dropped as overcomplicating things) is something I'm not sure of.

Reply:
One option is to, instead of renaming the wheel, bake the python tag into the cache directory path instead. Though I don't really have a strong preference.